lib/ruby_lsp/utils.rb

```diff
@@ -26,6 +28,22 @@ module RubyLsp
     String,
   )
 
+  module_function
+
+  sig { params(source: String, char_position: Integer, encoding: Encoding).returns(Integer) }
+  def convert_to_byte_offset_position(source, char_position, encoding)
```

Comment:
I was unsure whether to implement this in `utils.rb`.
Please let me know if you think there is a more appropriate place.